buildalias/buildalias.go

```diff
@@ -0,0 +1,201 @@
+package main;
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type GameServerBuild struct {
+	BuildId string
+	Weight int
+}
+
+func main() {
+	args := os.Args;
+	fileName := "thundernetes-buildalias.yaml";
+	helpMessage := "\n\tThundernetes build alias feature\n\n" +
+	"1) create - generates a new build alias with the name provided." +
+	" Each alias is made of one or more pairs of build id and weights (priority of assignment for allocation). Template for using:" +
+	"\n\n\tbuildalias create <alias-name> <buildId1> <weight1> <buildId2> <weight2> ... <buildIdN> <weightN>\n\n" +
+	"2) allocate - create a new game server instance for the build alias and session id provided. Template for using\n\n\t" +
+	"buildalias allocate <alias-name> <sessionId>\n\n" +
+	"3) help - Displays this message and exit";
+
+	if len(args) == 1 {
+		fmt.Println(helpMessage);
+		return;
+	}
+
+	switch(args[1]) {
+
+	case "create":
+		CreateBuildAlias(args, fileName);
+
+	case "allocate":
+		AllocateForBuildAlias(args, fileName);
+
+	case "help":
+		fmt.Println(helpMessage);
+
+	default:
+		fmt.Println("Sorry, but the command "+ args[1] + " is not recognized.");
+	}
+}
+
+func CreateBuildAlias(args []string, fileName string) {
+	if len(args) < 5 {
+		log.Fatal("Not enough arguments were provided.");
+	}
+
+	fmt.Println("Creating the build alias: " + args[2]);
+
+	buildName := args[2];
+	aliasMap := map[string]map[string]int{};
+
+	aliasMap[buildName] = make(map[string]int);
+
+	for i := 3; i < len(args) - 1; i += 2 {
+		buildId := args[i];
+
+		if weight, err := strconv.Atoi(args[i+1]); err == nil {
+			aliasMap[buildName][buildId] = weight;
+		} else {
+			log.Fatal("Invalid weight " + args[i+1] + " provided for buildId: " + buildId + ". ", err);
+		}
+	}
+
+	yamlData, err := yaml.Marshal(&aliasMap);
+
+	if err != nil {
+		log.Fatal("Error while Marshaling. \n", err);
+	}
+
+	f, err := os.OpenFile(fileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644);
+
+	if err != nil {
+		log.Fatal("Error when creating the alias", err);
+	}
+
+	if _, err := f.Write(yamlData); err != nil {
+		f.Close();
+		log.Fatal(err);
+	}
+
+	if err := f.Close(); err != nil {
+		log.Fatal(err);
+	}
+
+	fmt.Println("Build alias " + buildName + " created successfully.");
+}
+
+func AllocateForBuildAlias(args []string, fileName string) {
+	if len(args) < 4 {
+		log.Fatal("Not enough arguments were provided.");
+	}
+
+	buildName := args[2];
+	sessionId := args[3];
+	var ipToAllocate string;
+
+	cmd := exec.Command("kubectl","get","svc","-n","thundernetes-system","thundernetes-controller-manager",

```

Comment:
we shouldn't require kubectl to execute this program, this might run somewhere where kubectl is not installed. We can have this as a parameter (or environment variable) when starting it

Reply:
now that I think of it, we can refer to the Service as thundernetes-controller-manager.thundernetes-system.svc.cluster.local
Yeah let's use that!

Reply:
so, should I call the service thundernetes-controller-manager.thundernetes-system.svc.cluster.local to generate the IP? Or how can I do it?

Reply:
you don't have to "generate" the IP, if you call http://thundernetes-controller-manager.thundernetes-system.svc.cluster.local/.... this will call the application via DNS. 

https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/